typescript/sdk/src/ism/HyperlaneIsmFactory.ts

```diff
@@ -90,20 +93,37 @@ export class HyperlaneIsmFactory extends HyperlaneApp<FactoryFactories> {
           );
           break;
       }
-      return this.deployMultisigIsm(chain, config);
+      contract = await this.deployMultisigIsm(chain, config);
     } else if (config.type === ModuleType.ROUTING) {
       this.logger(
         `Deploying Routing ISM to ${chain} for verifying ${Object.keys(
           config.domains,
         )}`,
       );
-      return this.deployRoutingIsm(chain, config);
+      contract = await this.deployRoutingIsm(chain, config);
     } else if (config.type === ModuleType.AGGREGATION) {
       this.logger(`Deploying Aggregation ISM to ${chain}`);
-      return this.deployAggregationIsm(chain, config);
+      contract = await this.deployAggregationIsm(chain, config, origin);
     } else {
       throw new Error(`Unsupported ISM type`);
     }
+
+    const moduleType = ModuleType[config.type];
+
+    if (!this.deployedIsms[chain]) {
+      this.deployedIsms[chain] = {};
+    }
+
+    if (origin) {
+      if (!this.deployedIsms[chain][origin]) {
+        this.deployedIsms[chain][origin] = {};
+      }
+      this.deployedIsms[chain][origin][moduleType] = contract;
+    } else {
+      this.deployedIsms[chain][moduleType] = contract;
+    }
```

Comment:
I know I contributed to this but can we add a few comments to clarify what's going on?
it's a bit messy

Reply:
Good point, added some comments - realised it's messy because the type of this object is messy tbh. We can worry about this later though